### PR TITLE
[New Sample] - Tab Meeting Recording and Transcript - NodeJs

### PR DESCRIPTION
**Pull request by Jegadeesh-MSFT**

### CODE REVIEW

**Comment by Jegadeesh-MSFT** [comment]
**Executed build pipeline**
image


URL: https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6691260240

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js [hunk 1]
```diff
@@ -0,0 +1,227 @@
...
+const axios = require('axios');
+const { DateTime } = require('luxon');
+const ENV_FILE = path.join(__dirname, '.env');
+const { Readable } = require('stream');
+require('dotenv').config({ path: ENV_FILE });
+
+const clientId = process.env.APP_REGISTRATION_ID;
+const clientSecret = process.env.CLIENT_SECRET;
+const graphScopes = ['https://graph.microsoft.com/User.Read'];
+
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:16]: remove extra line

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js [hunk 2]
```diff
@@ -0,0 +1,227 @@
...
+    const msalClient = new msal.ConfidentialClientApplication({
+        auth: {
+            clientId: clientId,
+            clientSecret: clientSecret
+        }
+    });
+    
+    let tenantId = jwt_decode(req.query.ssoToken)['tid']; //Get the tenant ID from the decoded toke
+
+    msalClient.acquireTokenOnBehalfOf({
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:84]: add this code in common function and use it

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js [hunk 3]
```diff
@@ -0,0 +1,227 @@
...
+                    }
+                  }
+                }
+    
+                CardResults.push(Obj);
+              }
+            }
+          }
+        }
+        return CardResults;
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:196]: add space before return

File: samples/tab-meeting-transcript-recording/nodejs/src/components/App.js [hunk 4]
```diff
@@ -0,0 +1,46 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+import React from 'react';
+import './App.css';
+import * as microsoftTeams from "@microsoft/teams-js";
+import { BrowserRouter,  Route, Routes } from "react-router-dom";
+
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/src/components/App.js:8]: remove extra line

File: samples/tab-meeting-transcript-recording/nodejs/src/components/meetingtranscriptrecording.js [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+                    setloginAdminAccount(true);
+                    setIsConsentButtonVisible(false);
+                    setIsLoginVisible(false);
+                    setLoading(false);
+                });
+        });
+    }
+
+    // Open stage view
+    const fetchrecordingtranscript = (subject, onlineMeetingId, transcriptsId, recordingId) => () => {
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/src/components/meetingtranscriptrecording.js:128]: add proper formatting

File: samples/tab-meeting-transcript-recording/nodejs/src/index.css [hunk 6]
```diff
@@ -0,0 +1,106 @@
...
+    padding: 0.3rem;
+    margin-left: 8rem;
+}
+.calendarSVG {
+    width: 1rem;
+}
+.txtMeetingTitle {
+    margin-left: 0.5rem;
+}
+.meetingDate {
```
Comment (Prithvi-MSFT) [inline samples/tab-meeting-transcript-recording/nodejs/src/index.css:40]: add space between css classes

**Review by Prithvi-MSFT — CHANGES_REQUESTED** [review]

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js (hunk 1 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:16]: fixed

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js (hunk 3 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:196]: Fixed

File: samples/tab-meeting-transcript-recording/nodejs/src/components/App.js (hunk 4 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/src/components/App.js:8]: Fixed

File: samples/tab-meeting-transcript-recording/nodejs/src/index.css (hunk 6 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/src/index.css:40]: fixed

File: samples/tab-meeting-transcript-recording/nodejs/src/components/meetingtranscriptrecording.js (hunk 5 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/src/components/meetingtranscriptrecording.js:128]: Fixed

File: samples/tab-meeting-transcript-recording/nodejs/api-server/server.js (hunk 2 above)
Comment (Jegadeesh-MSFT) [reply samples/tab-meeting-transcript-recording/nodejs/api-server/server.js:84]: Fixed

**Comment by Pawank-MSFT** [comment]
Executed build pipeline
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6731683390

**Review by Pawank-MSFT — APPROVED** [review]
looks correct, Approving!

**Review by Prithvi-MSFT — APPROVED** [review]

**Review by Wajeed-msft — APPROVED** [review]

**Comment by Pawank-MSFT** [comment]
@Jegadeesh-MSFT  Need to migrate into Teams Toolkit

**Comment by Pawank-MSFT** [comment]
Executed Build pipeline:
https://github.com/OfficeDev/Microsoft-Teams-Samples/actions/runs/6744647628

**Review by Pawank-MSFT — APPROVED** [review]
looks correct, Approving!

**Pull request merged**